### ISSUE TITLE
Implement location/role based AccessChecker

### ISSUE THREAD

**Issue by vivekmittal07** [google/fhir-gateway#103]

**Comment by bashir2** [comment]
Thanks @vivekmittal07 for filing this; do you mind separating this to at least two (or maybe more) issues to make them more fine-grained. For example, the requirement/design doc can be a separate issue and based on that we can have 2-3 other tickets for the implementation. Please also add a little bit of context to the main issue description.

**Comment by jjtswan** [comment]
Marking as new, since this is blocked on #105.

**Comment by bashir2** [comment]
Moving to backlog until we clarify what the plugin should look like (and it is generic enough).

**Comment by bashir2** [comment]
Discussing this further with @vivekmittal07 and the team, it seems it is unlikely to implement something that is generic enough to be used out-of-the-box by any implementer. So this would serve only as another example. We decided this is not a Beta launch blocker.

**Comment by jjtswan** [comment]
Providing something usable out-of-the-box, or at least that was a potential
stretch to if possible while providing a meaningful example that developers
could look at as a reference example for their own policies - which was the
actual main goal

Without this, I think it will be very hard for developers to grasp and see
here.

Do we have such an example today? And if not, what would be an alternate,
meaningful proposal after the research and investigation that has already
been done?

**Comment by vivekmittal07** [comment]
There are 2 possible goals for this -
1. Enable a partner to adopt FHIR Gateway by implementing an access checker which is generic and meets their access control requirements.
2. Build a sample Role/Location based access checker which may not solve a partner's use-case but can provide a good reference to start. Best case is if they are able to modify a few things to customize this based on their requirement. 

We as a group think option 1 is more useful and also this will help with the FHIR Gateway adoption. 
For option 2 we will make assumptions which might not be valid for a realistic usecase. Also we already have good access checkers which show how someone can consume data from Keycloak and read FHIR data for making access decision.

Action items for beta
- writeup design sketch of something that IPRD could/would use.
- Collect feedback from the FHIR community on the access control model.
- Add a design doc on github explaining how someone can implement Role/Location based access control model

Post beta, it would be good to implement this to see if we can get IPRD to start using this.
